### PR TITLE
[PM-9008] factor generator-extensions into separate libraries

### PR DESCRIPTION
**Pull request by audrey-jensen**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-9008

## 📔 Objective

Improve import usability of password generator extension modules by factoring them into `generator-navigation`, `generator-history`, and `generator-legacy`. 

The core issue is that modules require all imports to be from the root namespace. `generator-extensions` was written with path-based interfaces in mind (as one does with `@bitwarden/common`). This caused its exports to require "dot notation" to address, which is incompatible with Angular's dependency injectors.

Note: This ticket cannot be tested; it is not yet integrated with the clients.

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/tools/generator/extensions/legacy/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "license": "GPL-3.0",
+  "scripts": {
+    "clean": "rimraf dist",
+    "build": "npm run clean && tsc",
+    "build:watch": "npm run clean && tsc -watch"
+  },
+  "dependencies": {
+    "@bitwarden/common": "file:../../../common",
+    "@bitwarden/generator-core": "file:../core"
+  }
```
Comment (djsmith85) [inline libs/tools/generator/extensions/legacy/package.json:23]: `@bitwarden/generator-legacy` also depends on `@bitwarden/generator-history` and `@bitwarden/generator-navigation`

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Nice work on the new libs. Changes are looking good just two small things and then we are ready to go.

[jest.config.js](https://github.com/bitwarden/clients/blob/main/jest.config.js) needs to be extended as done with generator-core

File: libs/tools/generator/extensions/legacy/package.json (hunk 1 above)
Comment (audrey-jensen) [reply libs/tools/generator/extensions/legacy/package.json]: I suspected I'd forgotten dependency updates somewhere. Thanks!

Addressed in 3737387.

**Comment by audrey-jensen** [comment]
> [jest.config.js](https://github.com/bitwarden/clients/blob/main/jest.config.js) needs to be extended as done with generator-core

@djsmith85 - addressed by [45a4b1a](https://github.com/bitwarden/clients/pull/9724/commits/45a4b1a3a381d71f926e8b2235b14d79284e5fca)

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**